idaes/core/scaling/custom_scaler_base.py

```diff
@@ -370,16 +428,28 @@ def scale_constraint_by_default(self, constraint, overwrite: bool = False):
         Returns:
             None
         """
-        sf = self.get_default_scaling_factor(constraint)
-        if sf is not None:
-            self.set_constraint_scaling_factor(
-                constraint=constraint, scaling_factor=sf, overwrite=overwrite
-            )
-        else:
-            _log.debug(
-                f"Could not set scaling factor for {constraint.name}, "
-                f"no default scaling factor set."
-            )
+        if not isinstance(constraint, ConstraintData):
+            raise TypeError(f"{constraint} is not a constraint (or is indexed).")
+        self._scale_component_by_default(component=constraint, overwrite=overwrite)
+
+    def get_expression_nominal_value(self, expression):
```

Comment:
Is there not a better way to combine/differentiate the three nominal value functions?

Reply:
We have `get_expression_nominal_values` deprecated, and it can be removed in the next release. The difference between `get_sum_terms_nominal_values` and `get_expression_nominal_value` is that the former returns a list while the latter returns a float.